### PR TITLE
Upload file and photo. Support clipboard upload.

### PR DESCRIPTION
**Pull request by sarg**

### CODE REVIEW

File: telega-msg.el [hunk 1]
```diff
@@ -485,9 +485,13 @@ If MARKDOWN is non-nil then format TEXT as markdown."
                              :parse_mode (list :@type "textParseModeMarkdown")))
                     (list :@type "formattedText"
                           :text text :entities []))))
-    (list :@type "inputMessageText"
-          :text fmt-text
-          :clear_draft t)))
+
+    (if (string-match "^photo:\\(.+\\)$" text)
```
Comment (zevlg) [inline telega-msg.el]: This is limiting user's input, how he could then send "photo:xxxx" as plain text ?

File: telega-chat.el [hunk 2]
```diff
@@ -1048,6 +1048,18 @@ With prefix arg delete only for yourself."
   (telega-server--call (list :@type "generateChatInviteLink"
      :chat_id (or chat-id (plist-get telega-chatbuf--chat :id)))))
 
+(defun telega-chat-insert-clipboard-photo ()
+  "Save image in clipboard to file and paste link to it."
+  (interactive)
+  (if (not (fboundp 'x-get-selection))
+      (message "Not implemented")
```
Comment (zevlg) [inline telega-chat.el]: better just to assert for fboundp x-get-selection

File: telega-chat.el [hunk 3]
```diff
@@ -1048,6 +1048,18 @@ With prefix arg delete only for yourself."
...
+
+    (let* ((sel (x-get-selection 'CLIPBOARD 'image/png))
+           (tmp (when sel (make-temp-file "telega-photo" nil ".png"))))
+      (if tmp (progn (with-temp-file tmp (insert sel))
+                     (insert (message "photo:%s" tmp)))
+        (message "Clipboard doesn't contain image")))))
+
```
Comment (zevlg) [inline telega-chat.el]: Special syntax "photo:xxx" is bad idea, it limits user too much

File: telega-chat.el [hunk 4]
```diff
@@ -1048,6 +1048,18 @@ With prefix arg delete only for yourself."
...
 
+(defun telega-chat-insert-clipboard-photo ()
+  "Save image in clipboard to file and paste link to it."
+  (interactive)
+  (if (not (fboundp 'x-get-selection))
+      (message "Not implemented")
+
+    (let* ((sel (x-get-selection 'CLIPBOARD 'image/png))
+           (tmp (when sel (make-temp-file "telega-photo" nil ".png"))))
+      (if tmp (progn (with-temp-file tmp (insert sel))
```
Comment (zevlg) [inline telega-chat.el]: what about removing temporary file ?

File: telega-chat.el (hunk 3 above)
Comment (sarg) [reply telega-chat.el]: On the other hand it allows user to attach file by entering absolute path using autocomplete (e.g. `company-files`)

File: telega-msg.el (hunk 1 above)
Comment (sarg) [reply telega-msg.el]: Add a space at the beginning of line.

**Pull request closed without merging**